### PR TITLE
feat(nanovg): add blur and dropshadow support

### PR DESCRIPTION
**Pull request by FASTSHIFT**
## Summary

Add GPU-accelerated blur and drop-shadow rendering to the NanoVG draw backend using a two-pass separable Gaussian shader with FBO ping-pong.

This continues the work started in #10129 by @kisvegabor, reworked based on my architectural review in that PR. The key change is that the blur shader logic has been **pushed down into `nanovg_gl_utils.h`** as a reusable `nvgluBlurRegion()` API, keeping the LVGL porting layer thin and backend-portable.

## Motivation

- Widgets using `blur_backdrop`, `blur_radius`, and `drop_shadow_*` styles had no NanoVG backend support — they were either skipped or fell through to the software renderer (which cannot operate on FBO-backed layers).
- This PR enables frosted-glass effects and shape-accurate drop shadows on GPU-rendered UIs.

## Architecture

```mermaid
flowchart TD
    A["LVGL Draw Task<br/>(LV_DRAW_TASK_TYPE_BLUR)"] --> B
    B["lv_draw_nanovg_blur.c (~140 lines)<br/>• Clip area calculation<br/>• LVGL→GL coord conversion<br/>• Build NVGLUblurParams"] --> C
    C["nvgluBlurRegion() — nanovg_gl_utils.h<br/>• Shader compilation & caching<br/>• Gaussian kernel computation<br/>• FBO ping-pong (grow-only scratch)<br/>• GL state isolation"] --> D
    D["GL/GLES Backend<br/>(GL2 / GL3 / GLES2 / GLES3)"]

    style A fill:#e3f2fd,stroke:#1565c0
    style B fill:#e8f5e9,stroke:#2e7d32
    style C fill:#fff3e0,stroke:#e65100
    style D fill:#f3e5f5,stroke:#6a1b9a
```

### Before vs After (compared to #10129)

```mermaid
flowchart LR
    subgraph "#10129 (original)"
        direction TB
        X1["LVGL blur task"] --> X2["lv_draw_nanovg_blur.c<br/>~600 lines of raw GL"]
        X2 --> X3["glCreateShader / glUseProgram<br/>FBO mgmt / glDrawArrays"]
    end

    subgraph "This PR"
        direction TB
        Y1["LVGL blur task"] --> Y2["lv_draw_nanovg_blur.c<br/>~140 lines (adapter)"]
        Y2 --> Y3["nvgluBlurRegion()<br/>(NanoVG library API)"]
        Y3 --> Y4["GL backend internals"]
    end

    style X2 fill:#ffcdd2
    style Y2 fill:#c8e6c9
    style Y3 fill:#c8e6c9
```

## Changes

| File | Description |
|------|-------------|
| `src/libs/nanovg/nanovg_gl_utils.h` | New `nvgluBlurRegion()` API with shader, FBO management, kernel computation |
| `src/draw/nanovg/lv_draw_nanovg_blur.c` | New file — thin LVGL adapter translating blur tasks to `nvgluBlurRegion()` |
| `src/draw/nanovg/lv_draw_nanovg_private.h` | Add `blur_state` field and blur function declarations |
| `src/draw/nanovg/lv_draw_nanovg.c` | Register blur init/deinit/execute/evaluate, flush draws before layer switch |
| `src/draw/lv_draw.c` | Forward `drop_shadow_color`/`opa` through `blur_dsc` for NanoVG recolor |

### 1. Blur lives in NanoVG library layer (not LVGL)

As proposed in my review of #10129: raw GL code should not live in the LVGL porting layer. The blur shader, FBO management, and GL state transitions are encapsulated in `nanovg_gl_utils.h`, making future backend ports (Vulkan, Metal) straightforward.

### 2. Two quality modes

| Mode | Taps | Texture Fetches | Use Case |
|------|------|-----------------|----------|
| `NVGLU_BLUR_QUALITY_NORMAL` | 16 discrete | 32 per pass | Default, best visual quality |
| `NVGLU_BLUR_QUALITY_SPEED` | 8 bilinear | 16 per pass | Mobile/embedded, ~2× faster |

Speed mode uses [bilinear tap merging](http://rastergrid.com/blog/2010/09/efficient-gaussian-blur-with-linear-sampling/) — adjacent Gaussian samples are combined into single HW-filtered texture reads.

### 3. Grow-only scratch FBO

The scratch FBO used for ping-pong is only reallocated when a larger size is needed, eliminating per-frame allocation churn when multiple blur regions of varying sizes exist.

### 4. No fallback to SW renderer

The software renderer cannot operate on FBO-backed layers. When blur cannot be performed (no FBO support, radius exceeds 256px limit), the task is skipped with `LV_LOG_WARN` rather than falling back.

### 5. Drop shadow recolor baked into blur pass

NanoVG's image draw path cannot easily recolor a pre-existing FBO. The shadow color is forwarded through `blur_dsc.base.drop_shadow_color` and applied as a tint in the blur shader's final output (premultiplied alpha).

### Test case: Backdrop blur

The `lv_example_style_blur` example creates a translucent card with `blur_backdrop=true` and `blur_radius=18` over a gradient background with text. Expected result:
- Text under the card is visibly blurred (unreadable)
- Text outside the card remains sharp
- Card has a frosted-glass appearance with white tint

### Test case: Drop shadow

The `lv_example_style_drop_shadow` example applies a red drop shadow to an arc's indicator part with `drop_shadow_radius=16` and offset `(5, 10)`. Expected result:
- A soft red glow follows the arc's curved shape
- Shadow is offset from the arc indicator
- Shadow color matches `lv_color_hex(0xf44336)`

### Test case: Edge conditions

| Condition | Expected Behavior |
|-----------|-------------------|
| `blur_radius = 0` | No-op, returns immediately |
| `blur_radius > 256` | `LV_LOG_WARN`, task skipped |
| GL2 without FBO extension | `blur_state = NULL`, evaluate rejects task |
| Blur area fully clipped | No-op after intersection check |
| Multiple blur sizes per frame | Grow-only FBO handles without reallocation |

## Commits

1. `feat(draw/nanovg): add blur and dropshadow support` — Base implementation
2. `refactor(draw/nanovg): move blur shader into nanovg_gl_utils` — Architecture cleanup
3. `perf(draw/nanovg): optimize blur with bilinear tap reduction` — Initial bilinear attempt
4. `fix(draw/nanovg): add blur robustness checks` — Edge case handling
5. `fix(draw/nanovg): use discrete kernel for correct blur quality` — Fix visual artifacts
6. `perf(draw/nanovg): add bilinear blur for speed quality mode` — Bilinear as opt-in speed mode

## Related

- #10129 — Original PR by @kisvegabor (this PR supersedes it with architectural improvements)
- [Efficient Gaussian Blur with Linear Sampling](http://rastergrid.com/blog/2010/09/efficient-gaussian-blur-with-linear-sampling/) — Bilinear optimization reference

## Checklist

- [x] Code formatted with `scripts/code-format.py`
- [x] Builds clean with `-Wall -Wextra` (no new warnings)
- [x] Tested on GLES2 backend (PC simulator with SDL + EGL)
- [x] `lv_example_style_blur` renders correctly
- [x] `lv_example_style_drop_shadow` renders correctly
- [x] `lv_demo_widgets` renders without regressions
- [ ] Tested on GLES3 / GL3 backend
- [ ] Tested on actual embedded hardware
- [ ] Screenshot comparison tests (requires test infrastructure)

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/libs/nanovg/nanovg_gl_utils.h [hunk 1]
```diff
@@ -40,6 +40,36 @@ void nvgluBindFramebuffer(NVGLUframebuffer * fb);
 NVGLUframebuffer * nvgluCreateFramebuffer(NVGcontext * ctx, int w, int h, int imageFlags, int format);
 void nvgluDeleteFramebuffer(NVGLUframebuffer * fb);
 
+// ---- Blur utilities ----
+
+// Quality hint for blur operations
```
Comment (kisvegabor) [inline src/libs/nanovg/nanovg_gl_utils.h:45]: Please use `/**/` comments.

**Review by kisvegabor — COMMENTED** [review]
I like the idea of making it more generic.

File: src/libs/nanovg/nanovg_gl_utils.h (hunk 1 above)
Comment (FASTSHIFT) [reply src/libs/nanovg/nanovg_gl_utils.h:45]: I mainly followed nanovg's native comment style. If we want to completely rewrite nanovg in the future, we can use lvgl's `/**/` style (which I also prefer), but for now I suggest it's better to use `//` (to maintain a consistent style). What do you think?

**Review by kisvegabor — APPROVED** [review]
